### ISSUE TITLE
Crash in BasicEventSelection in MetaData handling from CBC

### ISSUE THREAD

**Issue by lawrenceleejr** [UCATLAS/xAODAnaHelpers#565]
Hi all,

We're seeing a nasty crash (like the one suggested in the docs here https://xaodanahelpers.readthedocs.org/en/latest/Issues.html#missing-metadata-information) when trying to use the metadata. When we turn on the m_useMetaData flag, and specify the derivation as SUSY1. 

In response to the docs, I made sure the derivation name was defined correctly and I've tried a few p-tags of SUSY1 and nothing seems to fix the crash. We can't just turn it off since SUSY1 has some significant skimming applied. 

Are others seeing this issue as well? The AR is Base,2.3.45. 

Thanks,
-Larry and Russell

Output here:
```
BasicEventSelection::B... INFO    Calling constructor
BasicEventSelection::B... INFO    Calling constructor
Running sample: mc15_13TeV.410000.PowhegPythiaEvtGen_P2012_ttbar_hdamp172p5_nonallhad.merge.DAOD_SUSY1.e3698_s2608_s2183_r7267_r6282_p2470
BasicEventSelection::h... INFO    Calling histInitialize
BasicEventSelection::h... INFO    Histograms initialized!
BasicEventSelection::f... INFO    Calling fileExecute
BasicEventSelection::f... INFO    Looking at DAOD made by Derivation Algorithm: SUSY1Kernel
TUnixSystem::DispatchS... ERROR   segmentation violation
```

Snippet of stack trace:
```
===========================================================
#6  0x00007f945e8e4834 in xAOD::CutBookkeeper_v1::nAcceptedEvents() const () from /cvmfs/atlas.cern.ch/repo/sw/ASG/AnalysisBase/2.3.45/RootCore/obj/x86_64-slc6-gcc49-opt/xAODCutFlow/lib/libxAODCutFlow.so
#7  0x00007f945d4a808c in BasicEventSelection::fileExecute() () at /afs/cern.ch/work/l/leejr/public/ZeroLeptonFactory/xAODAnaHelpers/Root/BasicEventSelection.cxx:272
#8  0x00007f946a275859 in EL::Worker::changeAlgState(EL::Worker::AlgInitState, EL::Worker::AlgExecState, EL::Worker::InputState) () from /cvmfs/atlas.cern.ch/repo/sw/ASG/AnalysisBase/2.3.45/RootCore/obj/x86_64-slc6-gcc49-opt/EventLoop/lib/libEventLoop.so
#3  XrdCl::JobManager::RunJobs (this=0x15bb2bb0) at /afs/slac/public/software/scalla/releases/v4.2.3/xrootd-4.2.3/src/XrdCl/XrdClJobManager.cc:146
#4  0x00007f9458834c99 in RunRunnerThread (arg=<value optimized out>) at /afs/slac/public/software/scalla/releases/v4.2.3/xrootd-4.2.3/src/XrdCl/XrdClJobManager.cc:33
#5  0x00007f948932caa1 in start_thread () from /lib64/libpthread.so.0
#6  0x00007f94889ee93d in clone () from /lib64/libc.so.6
```

**Comment by kratsg** [comment]
@kkrizka may want to look at this as well :)

**Comment by lawrenceleejr** [comment]
I'm running over the ttbar sample in SUSY1 with various p tags. For this particular test:

mc15_13TeV.410000.PowhegPythiaEvtGen_P2012_ttbar_hdamp172p5_nonallhad.merge.DAOD_SUSY1.e3698_s2608_s2183_r7267_r6282_p2470

**Comment by mmilesi** [comment]
Hi @lawrenceleejr ,

 the issue has to do with the name of the derivation algorithm.

I printed out the name of the available cutbookkeepers:

cbk name = SUSY1KernelSkim
cbk name = SUSY1KernelAug
cbk name = SUSY5KernelSkim
cbk name = SUSY5KernelAug

I didn't know there exist names that do not end with "Kernel" alone. But apparently here's a case.
I will make a fix for that, leaving up to the user to pass the full name of the DF algorithm used (check it with your DF experts!)

Marco

**Comment by kratsg** [comment]
@mmilesi - I would suggest adding some verbosity (since this is in initialize) to print out all the available cutbookkeeper names found. It would also be useful to protect against this somehow and error out stating they should look at the log and pick from one of the available cbk names?

**Comment by mmilesi** [comment]
That's right what I will add :wink:

**Comment by mmilesi** [comment]
Addressed in #567

**Comment by lawrenceleejr** [comment]
Hi Marco!

Thanks! I just tested 686427881df80c5c7ab783a77c0628adee35b77a and it works perfectly for me. 

Cheers,
-L
